pkg/odo/genericclioptions/context.go

```diff
@@ -0,0 +1,204 @@
+package genericclioptions
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/redhat-developer/odo/pkg/application"
+	"github.com/redhat-developer/odo/pkg/component"
+	"github.com/redhat-developer/odo/pkg/occlient"
+	"github.com/redhat-developer/odo/pkg/odo/util"
+	"github.com/redhat-developer/odo/pkg/project"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+// NewContext creates a new Context struct populated with the current state based on flags specified for the provided command
+func NewContext(command *cobra.Command) *Context {
+	return newContext(command, false)
+}
+
+// NewContextCreatingAppIfNeeded creates a new Context struct populated with the current state based on flags specified for the
+// provided command, creating the application if none already exists
+func NewContextCreatingAppIfNeeded(command *cobra.Command) *Context {
```

Comment:
NewContextWithAppCreate ?

Reply:
I think that `NewContextCreatingAppIfNeeded` is more explicit.